src/Service/CurriculumInventory/Export/XmlPrinter.php

```diff
@@ -96,156 +83,102 @@ public function print(array $inventory)
-        $rootNode->appendChild($institutionNode);
-        $institutionNameNode = $dom->createElementNS('http://ns.medbiq.org/member/v1/', 'm:InstitutionName');
-        $institutionNameNode->appendChild($dom->createTextNode($institution->getName()));
-        $institutionNode->appendChild($institutionNameNode);
-        $institutionIdNode = $dom->createElementNS(
-            'http://ns.medbiq.org/member/v1/',
-            'm:InstitutionID',
-            $institution->getAamcCode()
-        );
-        $institutionIdNode->setAttribute('domain', 'idd:aamc.org:institution');
-        $institutionNode->appendChild($institutionIdNode);
-        $addressNode = $dom->createElementNS('http://ns.medbiq.org/member/v1/', 'm:Address');
-        $institutionNode->appendChild($addressNode);
-        $streetAddressNode = $dom->createElementNS('http://ns.medbiq.org/address/v1/', 'a:StreetAddressLine');
-        $streetAddressNode->appendChild($dom->createTextNode($institution->getAddressStreet()));
-        $addressNode->appendChild($streetAddressNode);
-        $cityNode = $dom->createElementNS('http://ns.medbiq.org/address/v1/', 'a:City', $institution->getAddressCity());
-        $addressNode->appendChild($cityNode);
-        $stateNode = $dom->createElementNS(
-            'http://ns.medbiq.org/address/v1/',
-            'a:StateOrProvince',
-            $institution->getAddressStateOrProvince()
-        );
-        $addressNode->appendChild($stateNode);
-        $zipcodeNode = $dom->createElementNS(
-            'http://ns.medbiq.org/address/v1/',
-            'a:PostalCode',
-            $institution->getAddressZipcode()
-        );
-        $addressNode->appendChild($zipcodeNode);
-        $countryNode = $dom->createElementNS('http://ns.medbiq.org/address/v1/', 'a:Country');
-        $addressNode->appendChild($countryNode);
-        $countryCodeNode = $dom->createElementNS(
-            'http://ns.medbiq.org/address/v1/',
-            'a:CountryCode',
-            $institution->getAddressCountryCode()
-        );
-        $countryNode->appendChild($countryCodeNode);
+        $xw->startElement('Institution');
+        $xw->writeElement('m:InstitutionName', $institution->getName());
+        $xw->startElement('m:InstitutionID');
+        $xw->writeAttribute('domain', 'idd:aamc.org:institution');
+        $xw->text($institution->getAamcCode());
+        $xw->endElement(); // </m:InstitutionID>
+        $xw->startElement('m:Address');
+        $xw->writeElement('a:StreetAddressLine', $institution->getAddressStreet());
+        $xw->writeElement('a:City', $institution->getAddressCity());
+        $xw->writeElement('a:StateOrProvince', $institution->getAddressStateOrProvince());
+        $xw->writeElement('a:PostalCode', $institution->getAddressZipcode());
+        $xw->startElement('a:Country');
+        $xw->writeElement('a:CountryCode', $institution->getAddressCountryCode());
+        $xw->endElement(); // </a:Country>
```

Comment:
A++ for comment! Without it I had 0 clue what was going on here.

Reply:
agreed, comments of this nature are unfortunately necessary in order to keep track of things.  the fact that you have to "close your own tags" makes this more prone to error, so I'd prefer using DOM or SimpleXML over XMLWriter. You can't argue with the speed gains though, holy smokes.